### PR TITLE
TaskStatusCode handling

### PR DESCRIPTION
**Pull request by ewoutp**
This change adds well known constants for `StatusCode` field in the `Task` structure.

It also adds a `MapStatusToError` function to map the `StatusCode` of a given `Task` to an error.

### CODE REVIEW

**Comment by ewoutp** [comment]
@ZeissS RFR

File: tasks.go [hunk 1]
```diff
@@ -71,6 +71,15 @@ type TaskList []TaskListItem
 
 type TaskStatusCode int
 
+const (
+	StatusCodeNoError           = TaskStatusCode(0)
```
Comment (ZeissS) [inline tasks.go]: can we add a comment with a hint where these errors come from?

**Comment by ZeissS** [comment]
LGTM, but I would like to see a comment to the code where these constants come from

File: tasks.go (hunk 1 above)
Comment (ewoutp) [inline tasks.go]: Done

**Pull request merged**